app/questionnaire/value_source_resolver.py

```diff
@@ -100,12 +100,41 @@ def _resolve_list_item_id_for_value_source(
             else None
         )
 
+    def resolve_dynamic_answers(
```

Comment:
Does this need unit testing?

Reply:
Good idea - I decided it would probably be best to make this a private method but then update the existing test for value source resolver to cover this.

Annoyingly because MagicMock() is truthy, the is_answer_dynamic function keeps returning true in tests where its called by a Mock, so I had to mock out quite a few but all should be working now